src/instructlab/model/train.py

```diff
@@ -233,72 +400,9 @@ def get_files(pattern):
     else:
-        # Local
-        from ..mlx_explore.gguf_convert_to_mlx import load
-        from ..mlx_explore.utils import fetch_tokenizer_from_hub
-        from ..train.lora_mlx.convert import convert_between_mlx_and_pytorch
-        from ..train.lora_mlx.lora import load_and_train
-        from ..train.lora_mlx.make_data import make_data
-
-        if not skip_preprocessing:
-            try:
-                make_data(data_dir=data_dir)
-            except FileNotFoundError as exc:
-                click.secho(
-                    f"Could not read from data directory: {exc}",
-                    fg="red",
-                )
-                raise click.exceptions.Exit(1)
-
-        # NOTE we can skip this if we have a way ship MLX
-        # PyTorch safetensors to MLX safetensors
-        model_dir_local = model_dir.replace("/", "-")
-        model_dir_mlx = f"{model_dir_local}-mlx"
-        model_dir_mlx_quantized = f"{model_dir_local}-mlx-q"
-
-        if skip_quantize:
-            dest_model_dir = model_dir_mlx
-            quantize_arg = False
-        else:
-            dest_model_dir = model_dir_mlx_quantized
-            quantize_arg = True
-
-        if tokenizer_dir is not None and gguf_model_path is not None:
-            if not local:
-                tokenizer_dir_local = tokenizer_dir.replace("/", "-")
-                fetch_tokenizer_from_hub(tokenizer_dir, tokenizer_dir_local)
-
-            # no need to pass quantize_arg for now, script automatically detects if quantization is necessary based on whether gguf model is quantized or not
-            load(gguf=gguf_model_path, repo=tokenizer_dir, mlx_path=dest_model_dir)
-
-            for filename in os.listdir(model_dir_local):
-                shutil.copy(
-                    os.path.join(model_dir_local, filename),
-                    os.path.join(dest_model_dir, filename),
-                )
-            shutil.rmtree(model_dir_local, ignore_errors=True)
-
-        else:
-            # Downloading PyTorch SafeTensor and Converting to MLX SafeTensor
-            convert_between_mlx_and_pytorch(
-                hf_path=model_dir,
-                mlx_path=dest_model_dir,
-                quantize=quantize_arg,
-                local=local,
-            )
-
-        adapter_file_path = f"{dest_model_dir}/adapters.npz"
-        # train the model with LoRA
-
-        load_and_train(
-            model=dest_model_dir,
-            train=True,
-            data=data_dir,
-            adapter_file=adapter_file_path,
-            iters=iters,
-            save_every=10,
-            steps_per_eval=10,
-        )
-
-        # TODO copy some downloaded files from the PyTorch model folder
-        # Seems to be not a problem if working with a remote download with convert.py
+        # pull the training and torch args from the flags
+        # the flags are populated from the config as a base.
+        params = ctx.params
+        train_args = TrainingArgs(**params)
```

Comment:
Just need to make sure that the datapath being received here is valid. We can either:
- check the type of it e.g. file/dir and then give them a descriptive error
- OR we can just figure it out on our own

Reply:
```py
if is_dir(data_path):
  raise ValueError("It looks like the datapath you passed is either a directory or does not exist, please doublecheck the value of --data-path and try again")
```